### PR TITLE
issue #321 - adding new button on integration view to open camel k op…

### PR DESCRIPTION
**Pull request by bfitzpat**
…erator log

Signed-off-by: bfitzpat@redhat.com <bfitzpat@redhat.com>

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -134,6 +134,43 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
 		// add commands to create config-map and secret objects from .properties files
 		configmapsandsecrets.registerCommands();
 		
+		// create the integration view action -- open operator log
+		vscode.commands.registerCommand('camelk.integrations.openOperatorLog', async () => {
+			setStatusLineMessageAndShow(`Retrieving log for Apache Camel K Operator...`);
+			await kubectlutils.getNamedPodsFromKubectl(`camel-k-operator`).then( async (podNames) => {
+				await handleLogViaKubectlCli(podNames[0]).catch((error) => {
```
Comment (apupier) [inline src/extension.ts]: why is it taking the first one? What happens if there are several?

File: src/extension.ts [hunk 2]
```diff
@@ -134,6 +134,43 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
...
+		// create the context menu integration view action -- open kit log
+		vscode.commands.registerCommand('camelk.integrations.kitlog', async (node: TreeNode) => {
+			if (node && node.label) {
+				let integrationName : string = node.label;
+				setStatusLineMessageAndShow(`Retrieving log for Apache Camel K Integration kit...`);
+				await getIntegrationsListFromKamel(integrationName).then( async (result : string ) => {
+					if (result && result.length > 0) {
+						let lines : string[] = result.split("\n");
+						if (lines.length > 1) {
+							let secondLine = lines[1]; // "integration\tState\tkit-pod-name"
```
Comment (apupier) [inline src/extension.ts]: what means the comment?

**Comment by bfitzpat** [comment]
@apupier You realize that this was "work in progress" and not "ready for review", right?

File: src/extension.ts (hunk 1 above)
Comment (bfitzpat) [reply src/extension.ts]: there should only be one for a given namespace

Comment (bfitzpat) [reply src/extension.ts]: but I am asking the runtime guys to verify

File: src/extension.ts (hunk 2 above)
Comment (bfitzpat) [reply src/extension.ts]: "work in progress" is just that. :)

File: src/extension.ts [hunk 3]
```diff
@@ -134,6 +134,43 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
...
+				await handleLogViaKubectlCli(podNames[0]).catch((error) => {
+					utils.shareMessage(mainOutputChannel, `error: ${error} \n`);
+				});
+			}).catch( (err) => {
+				console.log(err);
+			});
+			hideStatusLine();
+		});
+		// create the context menu integration view action -- open kit log
+		vscode.commands.registerCommand('camelk.integrations.kitlog', async (node: TreeNode) => {
```
[the remark this answers is not in the document]
Comment (bfitzpat) [reply src/extension.ts]: yes, definitely some clean-up to be done moving things around

File: src/extension.ts (hunk 1 above)
Comment (bfitzpat) [reply src/extension.ts]: Seems based on info from the runtime folks that there will be at most 1, but may be none.

**Comment by bfitzpat** [comment]
>     * when removing the integration, the kit builder log should follow the same behavior than "normal" log. I.e. automatically close or put [Integration stopped]

That is incorrect. The kit builders will continue to run in the background until a `kamel reset` is called to clean them up. They do not get disposed. 

>     * what about adding a test checking that fro a given result of the cli, it can parse it and call the right function? it will help validating that the parsing is correct. it will allow to better understand what is expected output of the CLI.

I'm guessing you're talking about the kamel cli information parsing when I do a `kamel get` and parse to get the kit name? Can you clarify what you're suggesting here @apupier ?

**Comment by apupier** [comment]
I think would be more for another iteration, what do you think to separate the "remove" from the 2 "logs" contextual menu with a line?

**Comment by bfitzpat** [comment]
> I think would be more for another iteration, what do you think to separate the "remove" from the 2 "logs" contextual menu with a line?

We can do it, but the line is impossible to see regardless of the color theme

image

image

Not sure it's worth it.

**Comment by bfitzpat** [comment]
> * not important: the screenshots contains a "typo" for the workspace name: "camelk-exmples". If you're very courageous, you can change it otherwise that's fine with that

Can you tell me which image has the misspelling in it? 

> * optional (can be another PR): when clicking on "Open Apache Camel K Operator log" although the deployment has not been done yet. The error message is starting with 2 "Error" word. `Error Error from server (BadRequest): container "camel-k-operator" in pod "camel-k-operator-6bcf77ddf8-llmp4" is waiting to start: ContainerCreating`

Yes, I noticed that as well. Definitely good for a separate issue. https://github.com/camel-tooling/vscode-camelk/issues/325

> * works great, small little feedback inlined

Thanks.

**Comment by bfitzpat** [comment]
@apupier Corrections made except for double error error issue and menu divider, which we will investigate separately

**Comment by apupier** [comment]
> > I think would be more for another iteration, what do you think to separate the "remove" from the 2 "logs" contextual menu with a line?
> 
> We can do it, but the line is impossible to see regardless of the color theme
> 
> image
> 
> image
> 
> Not sure it's worth it.

I think there is a bug with VS Code on your system.

I can see the lines correctly:
Screenshot from 2020-02-26 15-11-11

**Comment by bfitzpat** [comment]
> I think there is a bug with VS Code on your system.

Could be an issue with this variety of linux. We can look into it later, I don't think it's critical for this PR.

**Comment by apupier** [comment]
> > I think there is a bug with VS Code on your system.
> 
> Could be an issue with this variety of linux. We can look into it later, I don't think it's critical for this PR.

ok, created https://issues.redhat.com/browse/FUSETOOLS2-235

**Pull request merged**